src/profiles_management/pmr/classes.py

```diff
@@ -0,0 +1,133 @@
+"""This package provides classes for validating and defining a PMR.
+
+The goal of these classes is to represent the PMR in a more
+Pythonic way, rather than a dict of vavlues.
+
+The classes should also provide some common helper methods for
+traversing the information of the PMR, but not functions for
+directly affecting the K8s cluster based on the PMR.
+
+This is similar to how the K8s python clients have classes
+for representing the objects and different methods for doing operations
+with them.
+"""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List
+
+import jsonschema
+
+from profiles_management.pmr.schema import RESOURCE_QUOTA_SCHEMA
+
+log = logging.getLogger(__name__)
+
+
+class UserKind(Enum):
+    """Class representing the kind of the user."""
+
+    USER = "user"
+    SERVICE_ACCOUNT = "service-account"
+
+
+class ContributorRole(Enum):
+    """Class representing the role of the user."""
+
+    ADMIN = "admin"
+    EDIT = "edit"
+    VIEW = "view"
+
+
+@dataclass
+class Contributor:
+    """Class representing which users should have access to a Profile."""
+
+    name: str
+    role: ContributorRole
+
+
+@dataclass
+class Owner:
+    """Class representing the owner field of a Profile."""
+
+    name: str
+    kind: UserKind
+
+
+@dataclass
+class Profile:
+    """Class representing a Profile and its Contributors."""
+
+    name: str
+    owner: Owner
+    resources: Dict[str, Any]
+    contributors: List[Contributor]
```

Comment:
> Are `contributors` part of the Kubeflow Profile always?

Contributors are orthogonal from a Profile CR. They are relevant to KFAM (CRUD) and the Central Dashboard.

> What happens if a Kubeflow Profile does not have contributors? If not, what should happen if contributors is set to `None` or `[]`?

All RoleBindings and AuthorizationPolicies, for that contributor, should be removed in the Profile namespace.

> Are contributors part of the Kubeflow Profile always? What happens if a Kubeflow Profile does not have contributors? If not, what should happen if contributors is set to None or []? Also, what should be the "empty" value?

An empty list `[]`

Reply:
IMO I'd keep this to be `[]` by default, and let pydantic also not allow `None` (current behavior). But do you think I"m missing something with this?

Reply:
Ah now I see what you mean and how it affects it. Yes, contributors and resources can be `None`,  since in some cases they could be omitted. 

I've pushed a commit to make those 2 properties optional. The following code holds now:
```python
from profiles_management.pmr import classes

p1 = classes.Profile(name='daniela-p1', owner=classes.Owner(name="daniela", kind=classes.UserKind.USER))
print(p1)
# name='daniela-p1' owner=Owner(name='daniela', kind=<UserKind.USER: 'user'>) resources=None contributors=[]
```

3ddc46b